storage_service/locations/models/package.py

```diff
@@ -752,6 +752,106 @@ def backlog_transfer(self, origin_location, origin_path):
         self.status = Package.UPLOADED
         self.save()
 
+    def index_file_data_from_aip_mets(self):
+        """
+        Attempts to read an Archivematica AIP METS file inside this
+        package, then uses the retrieved metadata to generate one entry in the
+        File table in the database for each file inside the package.
+
+        :raises StorageException: if the transfer METS cannot be found,
+            or if required elements are missing.
+        """
+        aip_dir_name = os.path.basename(os.path.splitext(self.full_path)[0])
+        relative_path = os.path.join(aip_dir_name, "data", "METS." + self.uuid + ".xml")
+
+        path_to_mets, temp_dir = self.extract_file(relative_path)
+
+        mw = metsrw.mets.METSWriter()
+        mw.fromfile(path_to_mets)
+
+        for fsentry in mw.all_files():
+            metadata = self._parse_file_metadata(fsentry)
+
+            if metadata is not None:
+                aip_file = File()
+                aip_file.file_type = File.AIP
+                aip_file.package = self
+                aip_file.source_id = metadata['uuid']
+                aip_file.origin = self.origin_pipeline.uuid
+                aip_file.name = os.path.join(aip_dir_name, fsentry.path)
+                aip_file.ingestion_time = mw.createdate
+                if 'format_name' in metadata:
+                    aip_file.format_name = metadata['format_name']
+                if 'size' in metadata:
+                    aip_file.size = int(metadata['size'])
+                if 'pronom_id' in metadata:
+                    aip_file.pronom_id = metadata['pronom_id']
+                if 'normalized' in metadata:
+                    aip_file.normalized = metadata['normalized']
+                if 'validated' in metadata:
+                    aip_file.validated = metadata['validated']
+                aip_file.save()
+
+        shutil.rmtree(temp_dir)
+
+    def _parse_file_metadata(self, fsentry):
+        """
+        Cycle through an FSEntry object's AMDsec subsections and consolidate
+        PREMIS object/event metadata.
+        """
+        metadata = None
+
+        if fsentry.path != 'None':
+            metadata = {}
+
+            # Get technical metadata
+            if len(fsentry.techmds):
+                techmd = fsentry.techmds[0]
+                premis_object = metsrw.premis.Object.parse(techmd.contents.document, False)
```

Comment:
What's the `False`? Please name the arguments, especially if they're something without context like a boolean.